L1Trigger/L1THGCal/plugins/ntuples/HGCalTriggerNtupleHGCDigis.cc

```diff
@@ -57,23 +75,90 @@ initialize(TTree& tree, const edm::ParameterSet& conf, edm::ConsumesCollector&&
     tree.Branch("hgcdigi_wafer", &hgcdigi_wafer_);
     tree.Branch("hgcdigi_wafertype", &hgcdigi_wafertype_);
     tree.Branch("hgcdigi_cell", &hgcdigi_cell_);    
+    tree.Branch("hgcdigi_cell_eta", &hgcdigi_cell_eta_);    
+    tree.Branch("hgcdigi_cell_phi", &hgcdigi_cell_phi_);    
+    tree.Branch("hgcdigi_cell_z", &hgcdigi_cell_z_);    
     tree.Branch("hgcdigi_data", &hgcdigi_data_);
-
+    tree.Branch("hgcdigi_isadc", &hgcdigi_isadc_);
+    tree.Branch("hgcdigi_istot", &hgcdigi_istot_);
+    if (is_Simhit_comp_) tree.Branch("hgcdigi_cell_simhit_energy", &hgcdigi_cell_simhit_energy_);
 }
 
 void
 HGCalTriggerNtupleHGCDigis::
 fill(const edm::Event& e, const edm::EventSetup& es)
 {
+ 
+    es.get<IdealGeometryRecord>().get("HGCalEESensitive", geom_ee);
+    es.get<IdealGeometryRecord>().get("HGCalHESiliconSensitive", geom_fh);
+    es.get<IdealGeometryRecord>().get("HGCalEESensitive",topo_ee);
+    es.get<IdealGeometryRecord>().get("HGCalHESiliconSensitive",topo_fh);
     edm::Handle<HGCEEDigiCollection> ee_digis_h;
-    edm::Handle<HGCHEDigiCollection> fh_digis_h;
-
     e.getByToken(ee_token_, ee_digis_h);
-    e.getByToken(fh_token_, fh_digis_h);
-
     const HGCEEDigiCollection& ee_digis = *ee_digis_h;
+    edm::Handle<HGCHEDigiCollection> fh_digis_h;
+    e.getByToken(fh_token_, fh_digis_h);
     const HGCHEDigiCollection& fh_digis = *fh_digis_h;
 
+    // retrieve simhit collections
+    std::unordered_map<uint32_t, double> simhits_ee;
+    std::unordered_map<uint32_t, double> simhits_fh;
+    
+    // sim hit association
+    if (is_Simhit_comp_) {
```

Comment:
Put all simhit association stuff in a separate function, called only if is_simhit_comp is true